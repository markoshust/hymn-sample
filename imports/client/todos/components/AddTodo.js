import React, { PropTypes } from 'react';
import { compose, withHandlers } from 'recompose';
import { connect } from 'react-redux';
import { add } from '../actions';

let input;

const AddTodo = ({ onSubmit }) => (
  <div>
    <form
      onSubmit={onSubmit}
    >
      <input ref={(node) => { input = node; }} />
      <button type="submit">
        Add Todo
      </button>
    </form>
  </div>
);

AddTodo.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};

const enhance = compose(
  connect(),
  withHandlers({
    onSubmit: ({
      dispatch,
    }) => (e) => {
      e.preventDefault();

      if (input.value.trim()) {
        dispatch(add(input.value));
        input.value = '';
      }
    },
  }),
);

export default enhance(AddTodo);
