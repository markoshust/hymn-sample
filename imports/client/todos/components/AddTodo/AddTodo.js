import React, { PropTypes } from 'react';
import { compose, withHandlers } from 'recompose';

let input;

const AddTodo = ({ onSubmit }) => (
  <div>
    <form onSubmit={onSubmit}>
      <input ref={(node) => { input = node; }} />
      <button type="submit">
        Add Todo
      </button>
    </form>
  </div>
);

AddTodo.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};

const enhance = compose(
  withHandlers({
    onSubmit: ({
      onSubmit,
    }) => (event) => {
      event.preventDefault();

      if (input.value.trim()) {
        onSubmit(input.value);
        input.value = '';
      }
    },
  }),
);

export default enhance(AddTodo);
