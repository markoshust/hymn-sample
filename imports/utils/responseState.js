const responseState = (response, ...state) => (
  response[state] ? response[state]() : state
);

export default responseState;
