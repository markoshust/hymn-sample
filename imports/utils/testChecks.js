import renderer from 'react-test-renderer';

const componentCheck = (component) => {
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

const renderCheck = (children) => {
  const component = renderer.create(children);

  return componentCheck(component);
};

export {
  componentCheck,
  renderCheck,
};
