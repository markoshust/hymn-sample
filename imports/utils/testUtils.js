import renderer from 'react-test-renderer';

const snap = (children) => {
  const component = renderer.create(children);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

export {
  snap, // eslint-disable-line import/prefer-default-export
};
