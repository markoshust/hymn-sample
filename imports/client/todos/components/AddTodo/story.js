import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import AddTodo from './AddTodo';

const stories = storiesOf('todos/AddTodo', module);

stories.add('Standard', () => (
  <AddTodo onSubmit={action('submit')} />
));
