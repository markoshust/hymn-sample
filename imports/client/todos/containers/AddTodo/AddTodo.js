import { connect } from 'react-redux';
import { todosAdd } from '../../actions';
import AddTodo from '../../components/AddTodo';

const mapDispatchToProps = dispatch => ({
  onSubmit: todo => dispatch(todosAdd(todo)),
});

export default connect(
  () => ({}),
  mapDispatchToProps
)(AddTodo);
