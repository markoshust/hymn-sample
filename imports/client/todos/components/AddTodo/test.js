import React from 'react';
import { renderCheck } from 'utils/testChecks';
import AddTodo from './AddTodo';

const onSubmit = jest.fn();

describe('<AddTodo />', () => {
  it('renders correctly', () => {
    renderCheck(
      <AddTodo onSubmit={onSubmit} />
    );
  });
});
