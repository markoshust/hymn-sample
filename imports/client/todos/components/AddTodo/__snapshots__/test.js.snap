exports[`<AddTodo /> renders correctly 1`] = `
<div>
  <form
    onSubmit={[Function anonymous]}>
    <input />
    <button
      type="submit">
      Add Todo
    </button>
  </form>
</div>
`;
